### PR TITLE
 K8SPSMDB-848: Remove formatted strings from log messages

### PR DESCRIPTION
**Pull request by inelpandzic**
[K8SPSMDB-848](https://jira.percona.com/browse/K8SPSMDB-848) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
In a couple of places, we use formatted log messages, breaking the whole structured logging thing we have.

**Cause:**
Misuse of our logging lib.

**Solution:**
Replace formatted messages with constant messages and proper use of KV pairs for the dynamic part of a log.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Are E2E tests passing?
- [ ] Are unit tests passing?
- [ ] Are linting tests passing?
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?
- [ ] Are OpenShift compare files changed for E2E tests (`compare/*-oc.yml`)?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported MongoDB version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Review by egegunes — APPROVED** [review]

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 1]
```diff
@@ -170,8 +170,8 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 			cr.Spec.Sharding.Mongos.Size = 0
 		} else {
 			if !cr.Spec.UnsafeConf && cr.Spec.Sharding.Mongos.Size < minSafeMongosSize {
-				log.Info(fmt.Sprintf("Mongos size will be changed from %d to %d due to safe config", cr.Spec.Sharding.Mongos.Size, minSafeMongosSize))
-				log.Info("Set allowUnsafeConfigurations=true to disable safe configuration")
```
Comment (hors) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:174]: Why do we decide to remove this message?

File: pkg/psmdb/pmm.go [hunk 2]
```diff
@@ -332,8 +332,7 @@ func AddPMMContainer(cr *api.PerconaServerMongoDB, secret *corev1.Secret, custom
 	}
 
 	if !cr.Spec.PMM.HasSecret(secret) {
-		log.Info(fmt.Sprintf(`Can't enable PMM: "%s" or "%s" with "%s" keys don't exist in the %s, or %s and %s secrets are out of sync`,
-			api.PMMAPIKey, api.PMMUserKey, api.PMMPasswordKey, cr.Spec.Secrets.Users, cr.Spec.Secrets.Users, api.InternalUserSecretName(cr)))
+		log.Info("Can't enable PMM, credentials are not setup correctly or secrets and internal secrets are our of sync")
```
Comment (hors) [inline pkg/psmdb/pmm.go:335]: As for me, the previous message was more informative. Now we say that something is wrong but the user can't understand what exactly.   

@inelpandzic In general if you want to change the log message you need to confirm it with @spron-in and with @fiowro.

File: pkg/apis/psmdb/v1/psmdb_defaults.go (hunk 1 above)
Comment (inelpandzic) [reply pkg/apis/psmdb/v1/psmdb_defaults.go:174]: If you look at the following log messages we have I think it is unnecessary. And for the second log line, which is obvious from our docs (if it is not it should be), I don't think we should put it in our logs.

File: pkg/psmdb/pmm.go (hunk 2 above)
Comment (inelpandzic) [reply pkg/psmdb/pmm.go:335]: Even though I'm not 100% certain and I wanted to see what you will say, but for me, I think this is enough, for the details about the correct setup user can reference our docs.

And why can't we change our logs, I think we should do it on our own, we are not breaking anything by changing them.

Comment (egegunes) [reply pkg/psmdb/pmm.go:335]: @hors I agree with Inel, why do we need approval from Sergey or Dmitriy for changing log messages? We can get their input but it should be our decision.

Having said that, I agree with @hors on this message. I didn't review the changes carefully enough. Previous one was informative. Does the user know what should they do with the new message? Docs help?

Comment (inelpandzic) [reply pkg/psmdb/pmm.go:335]: Like @hors said, current docs would probably not help. And like I said, I wasn't 100% sure about it and wanted your opinion and now I can agree that we need to enrich it.

Comment (inelpandzic) [reply pkg/psmdb/pmm.go]: [fdbcefb](https://github.com/percona/percona-server-mongodb-operator/pull/1118/commits/fdbcefbe35596827a3c780c7c020b17d520ff6e7)

**Review by egegunes — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/storage.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/one-pod.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/init-deploy.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/upgrade-consistency.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/cross-site-sharded.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/demand-backup.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/self-healing-chaos.log) |
| ignore-labels-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/ignore-labels-annotations.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/limits.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/scaling.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/monitoring-2-0.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/scheduled-backup.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/operator-self-healing-chaos.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/security-context.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/arbiter.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/rs-shard-migration.log) |
| demand-backup-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/demand-backup-sharded.log) |
| service-per-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/service-per-pod.log) |
| liveness | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/smart-update.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/version-service.log) |
| upgrade | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/upgrade.log) |
| upgrade-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/upgrade-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/users.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/pitr.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/data-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/non-voting.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/demand-backup-eks-credentials.log) |
| data-at-rest-encryption | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/data-at-rest-encryption.log) |
| pitr-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1118/804f6418/pitr-sharded.log) |
| We run 30 out of 30|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1118/commits/804f6418792732b189b2b5bde1d03560da8cac7e
image: `perconalab/percona-server-mongodb-operator:PR-1118-804f6418`

**Review by hors — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**